modules/imgcodecs/src/grfmt_webp.cpp

```diff
@@ -331,6 +387,115 @@ bool WebPEncoder::write(const Mat& img, const std::vector<int>& params)
     return size > 0;
 }
 
+bool WebPEncoder::writemulti(const std::vector<Mat>& img_vec, const std::vector<int>& params)
+{
+    if (img_vec.size() > 1)
+        CV_LOG_WARNING(NULL, "\nyou are trying to write " << img_vec.size() <<
+            " images.\nonly the first image will be written.\nuse imwriteanimation() function to write more images.\n");
+    return write(img_vec[0], params);
+}
+
+bool WebPEncoder::writeanimation(const Animation& animation, const std::vector<int>& params)
+{
+    CV_CheckDepthEQ(animation.frames[0].depth(), CV_8U, "WebP codec supports only 8-bit unsigned images");
+    int ok = 0;
+    int timestamp = 0;
+    const int width = animation.frames[0].cols, height = animation.frames[0].rows;
+
+    WebPAnimEncoderOptions anim_config;
+    WebPConfig config;
+    WebPPicture pic;
+    WebPData webp_data;
+
+    WebPDataInit(&webp_data);
+    if (!WebPAnimEncoderOptionsInit(&anim_config) ||
+        !WebPConfigInit(&config) ||
+        !WebPPictureInit(&pic)) {
+        CV_LOG_ERROR(NULL, "Library version mismatch!\n");
+        WebPDataClear(&webp_data);
+        return false;
+    }
+
+    int bgvalue = (static_cast<int>(animation.bgcolor[0]) & 0xFF) << 24 |
+        (static_cast<int>(animation.bgcolor[1]) & 0xFF) << 16 |
+        (static_cast<int>(animation.bgcolor[2]) & 0xFF) << 8 |
+        (static_cast<int>(animation.bgcolor[3]) & 0xFF);
+
+    anim_config.anim_params.bgcolor = bgvalue;
+    anim_config.anim_params.loop_count = animation.loop_count;
+
+    if (params.size() > 1)
+    {
+        if (params[0] == IMWRITE_WEBP_QUALITY)
+        {
+            config.lossless = 0;
+            config.quality = static_cast<float>(params[1]);
+            if (config.quality < 1.0f)
+            {
+                config.quality = 1.0f;
+            }
+            if (config.quality > 100.0f)
```

Comment:
@vrabaud could you take a look.. i think i should correct it as `if (config.quality >= 100.0f)`